jscomp/super_errors/super_typecore.ml

```diff
@@ -111,19 +111,12 @@ let print_expr_type_clash env trace ppf =
       @{<info>Moving these types above the component declaration@} should resolve this!@]\
     @]"
     (* This one above shouldn't catch any false positives, so we can safely not display the original type clash error. *)
-  else begin
-    if Super_reason_react.is_array_wanted_react_element trace then
+  else if Super_reason_react.is_component_spec_wanted_react_element trace then
```

Comment:
hmm, this does not compile, (not caught by the CI though, since it not snapshot)